addons/account/models/account_reconcile_model.py

```diff
@@ -103,6 +103,13 @@ def _compute_float_amount(self):
             except ValueError:
                 record.amount = 0
 
+    @api.onchange('model_id')
+    def _onchange_model_id_counterpart_type(self):
+        # New lines added for the creation of moves are more likely to require percentage
+        # of statement line instead of the default percentage of balance (residual amount)
+        if self.model_id.counterpart_type in ('sale', 'purchase'):
+            self.amount_type = 'percentage_st_line'
```

Comment:
```suggestion
amount_type = fields.Selection(
    selection=[
        ('fixed', 'Fixed'),
        ('percentage', 'Percentage of balance'),
        ('percentage_st_line', 'Percentage of statement line'),
        ('regex', 'From label'),
    ],
    required=True,
    store=True,
    precompute=True,
    compute='_compute_amount_type',
)

@api.depends('model_id.counterpart_type')
def _compute_amount_type(self):
    for line in self:
        if line.model_id.counterpart_type in ('sale', 'purchase'):
            line.amount_type = 'percentage_st_line'
        else:
            line.amount_type = line.amount_type or 'percentage'
```

you also need to add a constraint avoiding something else than percentage_st_line combined with a sale/purchase counterpart_type. This is because nothing prevents the user to write an invalid value through a custom script/server action/import of data

Reply:
Now we allow any amount_type to be used with sale/purchase as well :)